Scripts/RemediationScripts/Remediate-BackupVaultImmutability.ps1

```diff
@@ -0,0 +1,534 @@
+
+                Write-Host $([Constants]::SingleDashLine)
+                Write-Host "Error fetching Backup Vault resource: Resource ID: [$($_.ResourceId)]. Error: $($_)" -ForegroundColor $([Constants]::MessageType.Error)
+            }    
+        }    
+    }
+    else {
+        if ([String]::IsNullOrWhiteSpace($FilePath)) {
+            Write-Host "Fetching all Backup Vault(s) in Subscription: [$($context.Subscription.SubscriptionId)]" -ForegroundColor $([Constants]::MessageType.Info)
+            Write-Host $([Constants]::SingleDashLine)
+    
+            # Get all Backup Vault(s) in a Subscription
+            $backupVaultDetails = Get-AzDataProtectionBackupVault -ErrorAction Stop
+    
+            $backupVaultDetails = Get-BackupVaultDetails $backupVaultDetails
+        }
+        else {
+            if (-not (Test-Path -Path $FilePath)) {
+                Write-Host "ERROR: Input file: [$($FilePath)] not found. Exiting..." -ForegroundColor $([Constants]::MessageType.Error)
+                Write-Host $([Constants]::DoubleDashLine)	
+                return
+            }
+            Write-Host "Fetching all Backup Vault(s) from [$($FilePath)]..." -ForegroundColor $([Constants]::MessageType.Info)
+            Write-Host $([Constants]::SingleDashLine)
+            $BackupVaults = Import-Csv -LiteralPath $FilePath
+    
+            $validBackupVaults = $BackupVaults | Where-Object { ![String]::IsNullOrWhiteSpace($_.ResourceId) }
+    
+            $validBackupVaults | ForEach-Object {
+                $resourceId = $_.ResourceId
+    
+                try {                
+                    $backupVaultResource = Get-AzDataProtectionBackupVault -ResourceGroupName $_.ResourceGroupName -VaultName $_.ResourceName -ErrorAction SilentlyContinue
+                    $backupVaultDetails += Get-BackupVaultDetails $backupVaultResource
+                }
+                catch {
+                    Write-Host "Error fetching Backup Vault(s) resource: Resource ID:  [$($resourceId)]. Error: $($_)" -ForegroundColor $([Constants]::MessageType.Error)
+                }
+            }
+        }
+    }
+
+    $totalBackupVaults = ($backupVaultDetails | Measure-Object).Count
+
+    if ($totalBackupVaults -eq 0) {
+        Write-Host "No Backup Vault(s) found. Exiting..." -ForegroundColor $([Constants]::MessageType.Update)
+        Write-Host $([Constants]::DoubleDashLine)	
+        return
+    }
+
+    Write-Host "Found [$($totalBackupVaults)] Backup Vault(s)." -ForegroundColor $([Constants]::MessageType.Update)
+    Write-Host $([Constants]::SingleDashLine)
+    
+    # Backup Vault(s) where immutability is not locked
+    $backupVaultsWithoutImmutability = @()
+
+    Write-Host "Separating Backup Vault(s) for which immutability is not locked..." -ForegroundColor $([Constants]::MessageType.Info)
+ 
+    $backupVaultDetails | ForEach-Object {
+        $backupVault = $_
+        if ($_.ImmutabilityState.ToString().ToUpper() -ne "LOCKED") {
```

Comment:
NIT: use -ine for case insensitive comparision

Reply:
Removed ToUpper and used -ine for case insensitive comparision.